homework/hm.js

```diff
@@ -0,0 +1,114 @@
+// String 1. Consider the following string:
+const myString = "hello,this,is,a,difficult,to,read,sentence";
+console.log(myString);
+console.log(myString.length);
+const newStr = myString.split(",");
+console.log(newStr);
+const strStr = newStr.join(' ');
+console.log(strStr);
+// or 
+// myStr = myString.replace(/,/g, ' ');
+// console.log(myStr);
+
+// Sring 2.reverse
+const str = 'dlroW olleH';
+const arr = str.split('');
+console.log(arr);
+arr.reverse();
+console.log(arr);
+console.log(arr.join(''));
+
+// Arrays  Consider the following array:
+const favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+favoriteAnimals.push("turtle");
+ // new arry wil be ["blowfish", "capricorn", "giraffe","turtle"]
+console.log(favoriteAnimals);
+favoriteAnimals.splice(1,0,"cat");
+// new arry wil be ["blowfish","cat", "capricorn", "giraffe","turtle"]
+console.log(favoriteAnimals);
+favoriteAnimals.splice(3,1);
+console.log(favoriteAnimals);
+
+
+// Operators 1.Consider the following snippet:
+0 == '0'; // true
+0 == [];  // true
+"0" == [];  // false
+// 0 == '0'; // both values are number  0 . ---true
+// 0 == []; // 0 is number ,  if you coerce an empty array to a number, you get it's length - which is 0 --- true
+//  "0" ==  []; //JavaScript cannot coerce them to the same type - and one is falsy the other is not.
+
+// Operators 2
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+// i think all is true--- 1 === 1 , 2 === 2 , 3 === 3 
+// x == y --- true
+// x === y --- true 
+// z === y --- true
+// z == x --- true
+console.log(x == y); // false --why?
```

Comment:
this is `false` because javascript comparison on objects does a shallow compare, it looks for that `x` and `y` point to the same memory location, which in this case they don't.

Reply:
i don't understand  which two are in ( x y z)  same memory location?
thank you for your help.